### PR TITLE
Upgrade to version 4.0

### PR DESCRIPTION
**Pull request by GCorbel**
All is in the title.

### CODE REVIEW

File: app/assets/css/emojione-awesome.css [hunk 1]
```diff
@@ -1,5500 +1,39 @@
...
-  background-image: url("/images/emojione/svg/1f498.svg"); }
-
-.e1a-gift_heart {
-  background-image: url("/images/emojione/svg/1f49d.svg"); }
-
-.e1a-heart_decoration {
-  background-image: url("/images/emojione/svg/1f49f.svg"); }
-
-.e1a-peace {
-  background-im
```
Comment (jarthod) [inline app/assets/css/emojione-awesome.css]: we cannot use public URLs in the DD, are all the images up-to-date in the repo (I see no image change), can we use the relative version below everywhere?

**Comment by GCorbel** [comment]
I did a mistake yesterday on "emojione-awesome.css" I didn't saw I have to compile the file. I did it and finally saw than there is a compiled version on their repo : https://github.com/emojione/emojione/blob/master/extras/css/emojione-awesome.css .

About relative links, the first version is a link with the URL and there is an alternative with a relative link. For example, there is `.e1a-interrobang` and `.e1a-rel-interrobang`. Anyway, we don't use emojione-awesome.css in the DD.

About images, I did a wonderful bash script to check for missing files and it seems there is a lot of missing images. I check for a way to have them and I check if we can have them legally for free.

**Comment by GCorbel** [comment]
I checked for images and it's possible to download them for free, for the png version, on their website but we have to be subscribed and I'm not sure if we can use them in this emojione-rails legally.

Anyway, it still works because, in their script, it search images on their CDN : https://cdn.jsdelivr.net/emojione/assets/4.0/png/32/1f481-2640.png . There is no need to have them in the public directory now. Do you think it can cause problem with some internet restriction in some clients? Also, do you think we should remove the rake task and images from this repo because it is not used anymore?

**Comment by jarthod** [comment]
Yes it will cause problem because of internet restrictions, that's why I said "we cannot use public URLs in the DD" do we use PNG or SVG currently in DD? It seems with emojione 4 we can use it for free if it's PNG only and if we put an attribution link ☹

**Comment by nel** [comment]
We are using SVG version and the emojione version in this package were copy left. The version you just embedded is not and they now have "complex" licensing including restriction for redistribution etc ... So you can't let the Gem be a MIT marked one and I suggest we just don't publish that publicaly.

Moreover for our own use even if we pay a licence the price itself depends on the company size up to "undefined price" for 250+.

Question is do we really need the update if we don't want to support "racial" emojis ? In a year or two we can probably completly drop emojione as browser/OS support will be ok (outside linux)

**Comment by jarthod** [comment]
Why did you merge @GCorbel? this wasn't validated at all.

**Comment by GCorbel** [comment]
Arg, it's my mistake. I rollback.

**Comment by GCorbel** [comment]
@nel The new icons are not only for different version of skin colors. I think we can keep this MR as it is and handle the file versions in our app. We have options to serve PNGs and SVGs in the public directory.

**Comment by jarthod** [comment]
Even if we do it on our side we need a license for the new version, and as you saw it's not obvious, so do we really need to update?

**Pull request merged**